RIOT-OS/RIOT
# pkg/nimble: improve initialization and add host thread

### Contribution description
This PR makes NimBLE a little easier to use. It
- runs the host in a dedicated thread
- runs NimBLE's threads from `auto_init`
- adapts the example application accordingly

### Testing procedure
build and run `examples/nimble_gatt` to verify that the changes to not break anything

### Issues/PRs references
none
